src/Futhark/CLI/CompareBench.hs

```diff
@@ -0,0 +1,322 @@
+-- | @futhark compare-bench@
+module Futhark.CLI.CompareBench (main) where
+
+import Control.Exception (catch)
+import Data.Bifunctor (Bifunctor (bimap, second))
+import Data.ByteString.Lazy.Char8 qualified as LBS
+import Data.Either qualified as E
+import Data.List qualified as L
+import Data.Map qualified as M
+import Data.Text qualified as T
+import Data.Vector qualified as V
+import Futhark.Bench
+import Futhark.Util.Options (mainWithOptions)
+import Statistics.Sample qualified as S
+import System.Console.ANSI (hSupportsANSI)
+import System.IO (stdout)
+import Text.Printf (printf)
+
+-- | Record that summerizes a comparison between two benchmarks.
+data SpeedUp = SpeedUp
+  { -- | What factor the benchmark is improved by.
+    speedup :: Double,
+    -- | Memory usage.
+    memoryUsage :: M.Map T.Text Double,
+    -- | If the speedup was significant.
+    significant :: Bool
+  }
+  deriving (Show)
+
+-- | Terminal colors used when printing the comparisons. Some of these are not
+-- colors ways of emphasising text.
+data Colors = Colors
+  { -- | The header color.
+    header :: String,
+    -- | Okay color
+    okblue :: String,
+    -- | A second okay color
+    okgreen :: String,
+    -- | Warning color.
+    warning :: String,
+    -- | When something fails.
+    failing :: String,
+    -- | Default color.
+    endc :: String,
+    -- | Bold text.
+    bold :: String,
+    underline :: String -- Underline ^ text.
+  }
+
+-- | Colors to use for a terminal device.
+ttyColors :: Colors
+ttyColors =
+  Colors
+    { header = "\ESC[95m",
+      okblue = "\ESC[94m",
+      okgreen = "\ESC[92m",
+      warning = "\ESC[93m",
+      failing = "\ESC[91m",
+      endc = "\ESC[0m",
+      bold = "\ESC[1m",
+      underline = "\ESC[4m"
```

Comment:
We can tolerate this, but I'm wondering whether using the proper prettyprinting library (https://github.com/diku-dk/futhark/blob/master/src/Futhark/Util/Pretty.hs) would be better, as we would then be able to strip out these codes when producing e.g. file output.